pkg/operator/sync.go

```diff
@@ -341,26 +343,91 @@ func (optr *Operator) syncRenderConfig(_ *renderConfig) error {
 		caData[CA.File] = string(CA.Data)
 	}
 
-	err = optr.kubeClient.CoreV1().ConfigMaps("openshift-config-managed").Delete(context.TODO(), "merged-trusted-image-registry-ca", metav1.DeleteOptions{})
+	cm, err = optr.kubeClient.CoreV1().ConfigMaps("openshift-config-managed").Get(context.TODO(), "merged-trusted-image-registry-ca", metav1.GetOptions{})
 	if err != nil && !errors.IsNotFound(err) {
 		return err
 	}
-
 	cmAnnotations := make(map[string]string)
 	cmAnnotations["openshift.io/description"] = "Created and managed by the machine-config-operator"
-	_, err = optr.kubeClient.CoreV1().ConfigMaps("openshift-config-managed").Create(
-		context.TODO(),
-		&corev1.ConfigMap{
+	if err != nil && errors.IsNotFound(err) {
+		klog.Infof("creating merged-trusted-image-registry-ca")
+		_, err = optr.kubeClient.CoreV1().ConfigMaps("openshift-config-managed").Create(
+			context.TODO(),
+			&corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:        "merged-trusted-image-registry-ca",
+					Annotations: cmAnnotations,
+				},
+				Data: caData,
+			},
+			metav1.CreateOptions{},
+		)
+		if err != nil {
+			return err
+		}
+	} else {
+		klog.Info("merged-trusted-image-registry-ca still exists, checking contents")
+		cmMarshal, err := json.Marshal(cm)
+		if err != nil {
+			return err
+		}
+		newCM := &corev1.ConfigMap{
 			ObjectMeta: metav1.ObjectMeta{
 				Name:        "merged-trusted-image-registry-ca",
 				Annotations: cmAnnotations,
 			},
 			Data: caData,
-		},
-		metav1.CreateOptions{},
-	)
-	if err != nil {
-		return err
+		}
+		newCMMarshal, err := json.Marshal(newCM)
+		if err != nil {
+			return err
+		}
+
+		// check for deletedData
+		stillExists := false
+		for file, data := range cm.Data {
+			stillExists = false
+			// for each file in the configmap
+			// does it match ANY of the new files? if not, it is deleted.
+			for newfile, newdata := range caData {
+				if newfile == file && newdata == data {
+					stillExists = true
```

Comment:
Suggestion: maybe make this logic check for any deletions instead, since you want to break out of the loop immediately if any deletions happened.

This way you don't have to set the extra logic above and below

Reply:
also though, I basically am "immediately breaking when a deletion is found" because, we are not sure if a file is deleted until we have compared it to all of the other files in the new data. If I were to switch this to be newFile != file && newData != data, we could presumptuously be saying there were deletions when there weren't? I could be missing something

Reply:
Ah ok, I think the logic is fine as it is then, I think I mis-read something